File: Codeforces/contests/div3/998/B.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;

struct cow{
    ll id;
    vector<ll>  cards;
};

bool comp(cow a, cow b){
    return a.cards[0] < b.cards[0];
}

int main(){
    ll  T =1;
    cin >> T;
    while(T--){
        ll n, m;
        cin >> n >> m;
        vector<cow> matrix(n);
        for(int i = 0; i < n; ++i){
            cow C;
            vector<ll> v;
            for(int j = 0; j < m; ++j){
                ll temp;
                cin >> temp;
                v.push_back(temp);
            }
            C.id = i + 1;
            C.cards = v;
            matrix.push_back(C);
        }
        sort(matrix.begin(),matrix.end(), comp);
        bool flag = true;
        for(int i = 0; i < n; ++i){
            for(int j = 0; j < m-1; ++j){
                if(matrix[i].cards[j] > matrix[i].cards[j+1]){
                    flag = false;
                    break;
                }
            } 
        }
        if(!flag){
            cout << "-1\n";
        }else{
            for(int i = 0; i < n; ++i){
                cout << matrix[i].id << " \n"[i == n-1];
            }
        }
    }
    return 0;
}

File: Codeforces/contests/div3/998/B.py
t = int(input())
while t:
    t-=1
    n, m = map(int, input().split())
    cows = []
    for i in range(n):
        c = sorted(list(map(int, input().split())))
        cows.append([i+1, c])
    cows.sort(key=(lambda x: x[1][0]))
    flag = True
    for i in range(n):
        for j in range(m):
            if j == m-1 and i == n-1:
                continue
            elif i == n-1:
                if cows[i][1][j] > cows[0][1][j+1]:
                    flag = False
                    break
            else:
                if cows[i][1][j] > cows[i+1][1][j]:
                    flag = False
                    break
                
    if not flag:
        print(-1)
    else:
        print(*[id for id, cards in cows])
